docs/specs/meta-clients/web3modal/api.md

```diff
@@ -1,142 +1,248 @@
+  "entries": number, // Required - Number of entries per page (e.g., `100`)
+  "search": string | null, // Optional search term (e.g., `MetaMask`)
+  "include": string | null, // Optional comma-separated list of wallet IDs to include
+  "exclude": string | null, // Optional comma-separated list of wallet IDs to exclude
+  "chains": string | null, // Optional comma-separated list of chain identifiers (e.g., `eip155:1`)
+  "platform": string | null // Optional platform filter (`ios` or `android`)
 }
+```
+
+#### Response
 
-interface Response {
-  count: number;
-  data: Array<{
-    id: string;
-    name: string;
-    homepage: string;
-    image_id: string;
-    order: number;
-    mobile_link: string | null;
-    desktop_link: string | null;
-    webapp_link: string | null;
-    app_store: string | null;
-    play_store: string | null;
-    injected:
-      | {
-          namespace: string;
-          injected_id: string;
-        }[]
-      | null;
-  }>;
+```typescript
+{
+  "count": number,
+  "data": [
+    {
+      "id": string,
+      "name": string,
+      "homepage": string,
+      "image_id": string,
+      "order": number,
+      "desktop_link": string | null,
+      "webapp_link": string | null,
+      "app_store": string | null,
+      "play_store": string | null,
+      "mobile": {
+        deep_link: string | null,
+        universal_link: string | null
+      }
+      "injected": [
+        {
+          "namespace": string,
+          "injected_id": string
+        }
+      ] | null
+    }
+  ]
 }
 ```
 
-### GET `/getWalletImage/:wallet_id`
+### `GET /v2/ios-data`
 
-Returns image logo for requested `wallet_id`. See [types](./types.md).
+Returns a list of iOS-facing wallet data, including the public Explorer ID and iOS deep link schema. This allows apps to check exhaustively what wallets are installed.
 
-```ts
-interface Request {
-  headers: Headers;
-}
+#### Response
 
-type Response = Blob;
+```typescript
+{
+  "count": number,
+  "data": [
+    {
+      "id": string,
+      "ios_schema": string
+    }
+  ]
+}
 ```
 
-### GET `/public/getAssetImage/:asset_id`
+### `GET /v2/android-data`
 
-Returns image for public asset for requested `asset_id`, i.e network logo, token logo or connector logo.
+Returns a list of Android-facing wallet data, including the public Explorer ID and Android App ID. This allows apps to check exhaustively what wallets are installed.
 
-```ts
-interface Request {
-  headers: Headers;
-}
+#### Response
 
-type Response = Blob;
+```typescript
+{
+  "count": number,
+  "data": [
+    {
+      "id": string,
+      "android_app_id": string
+    }
+  ]
+}
 ```
 
-#### Known Static Asset IDs
+### `GET /v2/wallet-image/{size}/{wallet_id}`
+
+Returns the image logo for the specified wallet ID.
+
+#### Route Parameters
+
+- `wallet_id`: The unique identifier for the wallet
+- `size`: The predefined image size, described as `{width}x{height}` (see [Known Image Sizes](#known-image-sizes))
```

Comment:
As a follow up we should document the process for generating new image sizes.

One thing to be aware of, is that with the current approach we're forcing the API consumers to use generic sizes and adjust on their end. So on paper, it means less requests since instead of requesting the same image in 2 different sizes, they would simply scale down the generic image.

With the new approach, if AppKit requests the same image in 56x56 and then in 100x100 it'll result in 2 times more requests and potentially a higher bandwidth consumption.

This is based on the assumption that AppKit requests the same logo in 2 different sizes so we can ignore the concern if it's not/won't be the case.
 
cc @chris13524

Reply:
> So on paper, it means less requests since instead of requesting the same image in 2 different sizes, they would simply scale down the generic image.

It _is_ fewer requests, however in real conditions... usually the smaller image is requested first, and then 5% of the time the larger image is requested (for example picking a wallet from the list). Furthermore, 100x100=10k vs 56x56=3,136 so it's quadratic improvement to use the smaller image. This results in very large improvements to bandwidth with the cost of 1 additional network request. E.g. requesting 10 3k images and 1 10k image is better than requesting 10 10k images (40k vs 100k). Bigger improvement with more images.

Reply:
Thanks for clarifying and clearing this up 🙏